src/main/java/uk/gov/companieshouse/api/interceptor/InternalUserInterceptor.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.companieshouse.api.interceptor;
+
+import java.io.IOException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+import uk.gov.companieshouse.api.util.security.AuthorisationUtil;
+import uk.gov.companieshouse.api.util.security.EricConstants;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+/**
+ * Pre handle method to authenticate the request before it reaches the controller by checking if
+ * the CH API key is valid. The request is also checked to see if the user is in a role which can
+ * make internal calls.
+ */
+@Component
+public class InternalUserInterceptor extends HandlerInterceptorAdapter {
+    
+    private static final Logger LOG = LoggerFactory.getLogger(String.valueOf(InternalUserInterceptor.class));
```

Comment:
Can you check with @jsimons1989 about the use of initialising the Logger with this class. An alternative would be to have the logger as an instance variable and passing in in on initialisation from the calling project (this might be better for correlation)